### ISSUE TITLE
Replace EntityManager::merge with EntityManager::persist in Doctrine DataFixtures

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#7184]
# Problem description

The EntityManagerInterface::merge is deprecated and we should instead of `merge` use `persist`.

Following fixtures should be updated:

 - [x] https://github.com/sulu/sulu/blob/2.4/src/Sulu/Bundle/MediaBundle/DataFixtures/ORM/LoadMediaTypes.php
 - [x] https://github.com/sulu/sulu/blob/2.4/src/Sulu/Bundle/ContactBundle/DataFixtures/ORM/LoadDefaultTypes.php  
 - [x] https://github.com/sulu/sulu/blob/2.5/src/Sulu/Bundle/SecurityBundle/DataFixtures/ORM/LoadSecurityTypes.php
 - [x] https://github.com/sulu/sulu/blob/2.5/src/Sulu/Bundle/MediaBundle/DataFixtures/ORM/LoadCollectionTypes.php

This change can also target the 2.4 branch.

# Proposed solution

## Why was `merge` over `persist` used

Fixtures should only be loaded once so if the MediaType image already exists it should not created again. Also we are forcing the ids as there are some logic based on them, without merge currently a unique key constraint error would appear.

Like merge so we need first check if the entity already exists, if it does we just do nothing.

## How to use persist

The DataFixtures gives us a ObjectManager not a EntityManager so we only have `find` available but we can use that to check if the fixture was already loaded:

```diff
+        if (!$manager->find(MediaType::class, 1)) {
             $mediaDocument = new MediaType();
             $mediaDocument->setId(1);
-            $mediaDocument = $manager->merge($mediaDocument);
+            $mediaDocument = $manager->persist($mediaDocument);
             $mediaDocument->setName('document');
+        }
```

Testing this running `bin/console sulu:build dev` multiple times should not error.

## How to get started (for beginners)

1. Clone the Sulu Repository

```bash
git clone git@github.com:sulu/sulu.git
```

2. Checkout the 2.6 branch and create a new branch:

```bash
cd sulu
git checkout 2.6
git checkout -b feature/replace-merge-fixtures
```

3. Start the database

Configure the used database in the `.env.local` file. Supported are MySQL or PostgreSQL a docker-compose.yml is available in:  https://github.com/sulu/skeleton

```env
DATABASE_URL="mysql://root:ChangeMe@127.0.0.1:3306/su_myapp?serverVersion=8.0.27&charset=utf8mb4"
```

```bash
docker compose up
```

4. Implement the issue

See above (How to use persist)

5. Test changes

For this specific issue `bin/console sulu:build dev` is the easist way to test if the change works. Functional tests also use that fixtures.

6. Fork sulu/sulu repository

```bash
git remote add your git@github.com:<your_github_username>/sulu.git
```

7. Create pull request

```bash
git add -A
git commit -m "Replace ..."
git push your <your_branch>
```

**Comment by shochdoerfer** [comment]
Just to clarify: In the problem description the "2.4 branch" is mentioned, in the "How to get started" section the 2.6 branch is mentioned. Which one is it?

**Comment by alexander-schranz** [comment]
We could already target the 2.6 branch so there not longer deprecations appear but required it is for 2.6.

**Comment by shochdoerfer** [comment]
Will take care then, give me a few days.

**Comment by alexander-schranz** [comment]
fixed in #7194
